service/replay_test.go

```diff
@@ -91,6 +91,67 @@ func TestReplayCache_Archive(t *testing.T) {
 	}
 }
 
+func TestReplayCache_Resize(t *testing.T) {
+	t.Run("Smaller", func(t *testing.T) {
+		cache := &ReplayCache{
+			capacity: 5,
+			active: map[uint32]empty{
+				1: {}, 2: {}, 3: {}, 4: {}, 5: {},
+			},
+			archive: map[uint32]empty{
+				6: {}, 7: {}, 8: {}, 9: {}, 10: {},
+			},
+		}
+
+		cache.Resize(3)
+
+		if cache.capacity != 3 {
+			t.Errorf("Expected capacity to be 3, got %d", cache.capacity)
+		}
+		if len(cache.active) != 3 {
+			t.Errorf("Expected active handshakes length to be 3, got %d", len(cache.active))
+		}
+		if len(cache.archive) != 3 {
+			t.Errorf("Expected archive handshakes length to be 3, got %d", len(cache.active))
+		}
+	})
```

Comment:
Can you add tests that demonstrates that you can still detect the salts when you shrink or grow?

Reply:
Done.